ami/utils.go

```diff
@@ -38,14 +38,28 @@ func command(action string, id string, v ...interface{}) ([]byte, error) {
 }
 
 func send(ctx context.Context, client Client, action, id string, v interface{}) (Response, error) {
+	if id == "" {
+		id, _ = GetUUID()
+	}
 	b, err := command(action, id, v)
 	if err != nil {
 		return nil, err
 	}
 	if err := client.Send(string(b)); err != nil {
 		return nil, err
 	}
-	return read(ctx, client)
+	for {
```

Comment:
why a looping here ? 

The ActionId, in case of included in the command, must have to be in the response.
I'd like to suggest returning a known error to be thread outside the library, for example: 

```Go
var (
  // ErrInvalidResponseActionID occurss when the response doesn't have the same action id
  ErrInvalidResponseActionID = errors.New("invalid response action id")
)

...
        response, err := read(ctx, client)
        if err != nil {
                return nil, err
        }
        if id == "" || id == response.Get("ActionID") {
                return response, err
        }
        return nil, ErrInvalidResponseActionID
```

Reply:
the AMI interface is asynchronous. the response returned by the "read" function may not be the response to the executed command, but some asynchronous event returned. An example of this is the Logout command, when you run it you will receive two messages from AMI, the response to the command and a "logout" event, depending on the order in which the messages arrives will fail.
the only way I could think of to avoid this is with a loop, waiting for the response to the actionid, but it makes the actionID mandatory